### DISCUSSION TITLE
Error occured while using some of the controlnet model

### DISCUSSION THREAD

**Discussion by magicalz** [azuritecoin/OnnxDiffusersUI#124] [Q&A] [answered]
First of all, thank you for the latest update of controlnet integration!

I've converted all 8 controlnet pretrained model to ONNX, with script conv_sdcnet_to_onnx.py, and I converted chilloutmix model to ONNX model with script conv_sd_to_onnx.py, adding the --controlnet_path "lllyasviel/sd-controlnet-canny", then I tested them both on console and UI. 
All the controlnet models work fine with testonnxcnet.py on console, some of the models(canny, depth, openpose, scribble) work fine on UI, but the others(hed, mlsd, normal, seg) show an error on UI with the error message "UnboundLocalError: local variable 'cnet_image' referenced before assignment" in console.

Another issue is some of the schedulers doesn't work with controlnet, DPMS_ms & DDIM work fine, but EulerA and Euler show an error message "TypeError: unsupported operand type(s) for *: 'numpy.ndarray' and 'Tensor'"

I don't have any idea about how to fix them.

![screenshot](https://user-images.githubusercontent.com/11372247/227464565-908a4166-ffce-4173-826c-a9dd92336877.JPG)

![screenshot1](https://user-images.githubusercontent.com/11372247/227464454-b299c9fa-db83-461b-b3c9-610aad3877ac.JPG)
![screenshot2](https://user-images.githubusercontent.com/11372247/227464498-6508d0bb-6fc1-4cf2-94d8-ab8ef88c087b.JPG)

**Comment by uchuusen** [comment, marked as answer]
Ah, sorry about that, I guess I forgot to test those schedulers, since I mainly use DPMS, DEIS, or UniPC. I'll have a look at it.
For the the error message about cnet_image, you can work around that by checking the Don't preprocess image checkbox, but you'll have to get the preprocessed images yourself. It currently only supports Canny, Openpose, Scribble and Depth by default, but I'll see about getting the others in too. Should be an easy fix.
Edit:
Okay, here's a fix for the scheduler bug if you want to patch it locally. Open up pipeline_onnx_stable_diffusion_controlnet.py under the modules subdir and edit line 175 so it looks like this:
`        latents = latents * np.float64(self.scheduler.init_noise_sigma)`
That should fix it. Will do a PR later.

**Reply by magicalz** [reply]
Thanks for your quickly reply!
Scheduler bug has been fixed.
Yeah, I can preprocess image with other tools, no hurry, please take your time to fix the cnet_image.
